llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -1001,7 +1001,10 @@ ModRefInfo BasicAAResult::getModRefInfo(const CallBase *Call,
 
   // Refine accesses to errno memory.
   if ((ErrnoMR | Result) != Result) {
-    if (AAQI.AAR.aliasErrno(Loc, Call->getModule()) != AliasResult::NoAlias) {
+    // Do not make any assumptions about errno on freestanding environments.
+    bool IsFreestanding = Call->getFunction()->hasFnAttribute("no-builtins");
```

Comment:
I don't think this check belongs here. I'd expect freestanding to only be relevant for isErrnoFunctionCall.